packages/@aws-cdk/aws-synthetics-alpha/lib/canary.ts

```diff
@@ -435,6 +420,31 @@ export class Canary extends cdk.Resource implements ec2.IConnectable {
     };
   }
 
+  private validateHandler(handler: string, runtime: Runtime) {
+    if (
+      runtime === Runtime.SYNTHETICS_PYTHON_SELENIUM_1_0 ||
+      runtime === Runtime.SYNTHETICS_NODEJS_PUPPETEER_3_0 ||
+      runtime === Runtime.SYNTHETICS_NODEJS_PUPPETEER_3_1 ||
+      runtime === Runtime.SYNTHETICS_NODEJS_PUPPETEER_3_2 ||
+      runtime === Runtime.SYNTHETICS_NODEJS_PUPPETEER_3_3
+    ) {
+      if (!handler.match(/^[0-9A-Za-z_\\-]+\.handler*$/)) {
+        throw new Error(`Canary Handler must be specified as \'fileName.handler\' for legacy runtimes, received ${handler}`);
+      }
+    } else if (runtime === Runtime.SYNTHETICS_NODEJS_PUPPETEER_3_4) {
+      if (!handler.match(/^[0-9A-Za-z_\\-]+\.[A-Za-z_][A-Za-z0-9_]*$/)) {
+        throw new Error(`Canary Handler must be specified either as \'fileName.handler\' or \'fileName.functionName\' for the \'syn-nodejs-puppeteer-3.4'\ runtime, received ${handler}`);
```

Comment:
where did you get the info that 3.4 has it's own specific rule? From the cfn docs it seems this isn't the case:

> For syn-python-selenium-1.1, syn-nodejs.puppeteer-3.4, and later runtimes, the handler can be specified as fileName.functionName , or you can specify a folder where canary scripts reside as folder/fileName.functionName .